### ISSUE TITLE
Some issues with peers

### ISSUE THREAD

**Issue by trajano** [GeekyAnts/NativeBase#4575]
### Description

peers  "react-native-safe-area-context", "react-native-svg" should not be version locked.  The @types should not be peer dependencies as they are not needed at runtime.

### CodeSandBox/Snack link

na

### Steps to reproduce

```
  "peerDependencies": {
    "@types/react": "*", // should not be here these are only used for development
    "@types/react-native": "*", // should not be here these are only used for development
    "react": "*",
    "react-dom": "*",
    "react-native": "*",
    "react-native-safe-area-context": "3.1.9", // should not be version locked or at least not locked to the point that 3.2 which is out cannot be used unless there's a real restriction.
    "react-native-svg": "12.1.0" //  // should not be version locked in case they release a new version and then nativebase becomes another limitting factor
  },
```

Please test with `npm` for the examples you're likely testing only on Yarn which means you will have less strict dependency checks unlike `npm`

### NativeBase Version

3.3.4

### Platform

- [ ] Android
- [ ] CRA
- [ ] Expo
- [ ] iOS
- [ ] Next

**Comment by surajahmed** [comment]
@trajano Thanks for reporting! We'll remove peer dependencies in the next release.

**Comment by BrandanKing** [comment]
@trajano This still doesn't look like it has been updated in the latest version on Native Base. This is starting to cause issues with Expo as Expo expects version 12.1.1 of react-native-svg.

**Comment by LadislavBohm** [comment]
This is also preventing from upgrading one of the related dependencies (react-native-safe-area-context or react-native-svg) and more importantly this is will fail for anyone getting started with bare react-native workflow.

This I believe is not fixable by patch-package.

**Comment by trajano** [comment]
@LadislavBohm correct, patch-package cannot be fix problems with package.json.  Thanks for testing ahead for me.

**Comment by mrkpatchaa** [comment]
@surajahmed any update on this one? 
Latest release still breaks when react-native-svg or react-native-safe-area-context is installed

**Comment by chrisbianca** [comment]
I have just submitted a PR (#4773) that addresses this issue

**Comment by surajahmed** [comment]
@trajano We've removed hardcoded peer dependencies. However, as @mrkpatchaa mentioned, there are some problems with the latest versions of `react-native-svg` and `react-native-safe-area-context`. We'll try to find some workaround and fix this in the coming release.

**Comment by surajahmed** [comment]
This has been resolved in version `3.4.9` or later. Thanks.

**Comment by rohankleem** [comment]
Hi, installing Native Base via npm to react native 0.69.4 Theme using 
"npm install native-base react-native-svg@12.1.1 react-native-safe-area-context@3.3.2" 
(docs at https://docs.nativebase.io/install-rn), results in error:
code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: myapp@0.0.1
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"18.0.0" from the root project
npm ERR!   peer react@"*" from native-base@3.4.12
npm ERR!   node_modules/native-base
npm ERR!     native-base@"*" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.2.0" from react-dom@18.2.0
npm ERR! node_modules/react-dom
npm ERR!   peer react-dom@"*" from native-base@3.4.12
npm ERR!   node_modules/native-base
npm ERR!     native-base@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

Thanks
